### PR TITLE
Add AdmissionCheckStrategy documentation

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind documentation

#### Which issue(s) this PR fixes:

Part of #1432 

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

### CODE REVIEW

File: site/content/en/docs/concepts/admission_check.md [hunk 1]
```diff
@@ -38,11 +38,71 @@ spec:
     name: prov-test-config
 ```
```
[the remark this answers is not in the document]
Comment (PBundyra) [reply site/content/en/docs/concepts/admission_check.md:40]: Done

File: site/content/en/docs/concepts/admission_check.md [hunk 2]
```diff
@@ -38,11 +38,71 @@ spec:
...
 
-### ClusterQueue admissionChecks
+### Usage
 
-Once defined, an AdmissionCheck can be referenced in the ClusterQueues' spec. All Workloads associated with the queue need to be evaluated by the AdmissionCheck's controller before being admitted.
+Once defined, an AdmissionCheck can be referenced in the ClusterQueue's spec. All Workloads associated with the queue need to be evaluated by the AdmissionCheck's controller before being admitted.
```
Comment (alculquicondor) [inline site/content/en/docs/concepts/admission_check.md:43]:
```suggestion
Once defined, an AdmissionCheck can be referenced in the [ClusterQueue](cluster_queue)'s spec. All Workloads associated with the queue need to be evaluated by the AdmissionCheck's controller before being admitted.
```

File: site/content/en/docs/concepts/admission_check.md [hunk 3]
```diff
@@ -38,11 +38,71 @@ spec:
...
 Similarly to `ResourceFlavors`, if an `AdmissionCheck` is not found or its controller has not marked it as `Active`, the ClusterQueue will be marked as Inactive.
 
+There are two ways of referencing AdmissionChecks in the ClusterQueue's spec:
+
+- `.spec.admissionChecks` - is the list of AdmissionChecks that will be run for all Workloads submitted to the ClusterQueue
+- `.spec.admissionCheckStrategy` - wraps the list of `AdmissionCheckStrategyRules` that give you more flexibility. It allows you to both run an AdmissionCheck for all Workloads or to associate an AdmissionCheck
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:49]:
```suggestion
- `.spec.admissionCheckStrategy` - wraps the list of `admissionCheckStrategyRules` that give you more flexibility. It allows you to both run an AdmissionCheck for all Workloads or to associate an AdmissionCheck
```

File: site/content/en/docs/concepts/admission_check.md [hunk 4]
```diff
@@ -38,11 +38,71 @@ spec:
...
+with a specific ResourceFlavor. To specify ResourceFlavors that an AdmissionCheck should run for use the `admissionCheckStrategyRule.onFlavors` field, and if you want to run AdmissionCheck for all Workloads, simply leave the field empty.
+
+See examples below:
+
+Using `.spec.AdmissionChecks`
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:54]:
```suggestion
Using `.spec.admissionChecks`
```

File: site/content/en/docs/concepts/admission_check.md [hunk 5]
```diff
@@ -38,11 +38,71 @@ spec:
...
+        nominalQuota: 9
+      - name: "memory"
+        nominalQuota: 36Gi
+      - name: "nvidia.com/gpu"
+        nominalQuota: 9
+  admissionChecks:
+  - sample-prov
+```
+
+Using `.spec.AdmissionCheckStrategy`
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:78]:
```suggestion
Using `.spec.admissionCheckStrategy`

```

File: site/content/en/docs/concepts/admission_check.md [hunk 6]
```diff
@@ -38,11 +38,71 @@ spec:
...
+  - coveredResources: ["cpu", "memory", "nvidia.com/gpu"]
+    flavors:
+    - name: "default-flavor"
+      resources:
+      - name: "cpu"
+        nominalQuota: 9
+      - name: "memory"
+        nominalQuota: 36Gi
+      - name: "nvidia.com/gpu"
+        nominalQuota: 9
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:73]:
```suggestion
<...>
```

Could you compress irrelevant fields?

File: site/content/en/docs/concepts/admission_check.md [hunk 7]
```diff
@@ -38,11 +38,71 @@ spec:
...
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:102]:
```suggestion
<...>
```

ditto.

File: site/content/en/docs/concepts/admission_check.md [hunk 8]
```diff
@@ -38,11 +38,71 @@ spec:
...
+kind: ClusterQueue
+metadata:
+  name: "cluster-queue"
+spec:
+  admissionCheckStrategy:
+    admissionChecks:
+    - name: "sample-prov"           # Name of the AdmissionCheck to be run
+      onFlavors: ["default-flavor"] # This AdmissionCheck will only run for Workloads that use default-flavor
+    - name: "sample-prov-2"
+      onFlavors: []                 # This AdmissionCheck will run for all Workloads regardless of a used ResourceFlavor
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:90]:
```suggestion
    - name: "sample-prov-2" # This AdmissionCheck will run for all Workloads regardless of a used ResourceFlavor
```

I guess that non-specified `onFlavors` indicates the same behavior.

File: site/content/en/docs/concepts/admission_check.md [hunk 9]
```diff
@@ -38,11 +38,71 @@ spec:
...
+Once defined, an AdmissionCheck can be referenced in the ClusterQueue's spec. All Workloads associated with the queue need to be evaluated by the AdmissionCheck's controller before being admitted.
 Similarly to `ResourceFlavors`, if an `AdmissionCheck` is not found or its controller has not marked it as `Active`, the ClusterQueue will be marked as Inactive.
 
+There are two ways of referencing AdmissionChecks in the ClusterQueue's spec:
+
+- `.spec.admissionChecks` - is the list of AdmissionChecks that will be run for all Workloads submitted to the ClusterQueue
+- `.spec.admissionCheckStrategy` - wraps the list of `AdmissionCheckStrategyRules` that give you more flexibility. It allows you to both run an AdmissionCheck for all Workloads or to associate an AdmissionCheck
+with a specific ResourceFlavor. To specify ResourceFlavors that an AdmissionCheck should run for use the `admissionCheckStrategyRule.onFlavors` field, and if you want to run AdmissionCheck for all Workloads, simply leave the field empty.
```
Comment (tenzen-y) [inline site/content/en/docs/concepts/admission_check.md:50]: Additionally, could you mention both fields (`admissionCheck` and `admissionCheckStrategy`) are not possible to be specified at the same time?

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**